===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ RegisterPreprocessorRuleOption:n$0 == { -2 } }) free(*(RegisterPreprocessorRuleOption:optionInfo)) at 5 (line 151, column 9),
===== Actual patch in diff format =====
--- a/table1/benchmarks/snort-2.9.13/src/dynamic-plugins/sp_preprocopt.c
+++ b/table1/benchmarks/snort-2.9.13/src/dynamic-plugins/sp_preprocopt.c
@@ -148,8 +148,13 @@ int RegisterPreprocessorRuleOption(
     optionInfo->otnHandler = otnHandler;
 
     ret = sfghash_add(sc->preproc_rule_options, optionName, optionInfo);
+    if (ret == 1)
+      free(optionInfo);
+
     if (ret != SFGHASH_OK)
     {
+        if (ret == -1 || ret == -2)
+          free(optionInfo);
         FatalError("Failed to initialize Preprocessor Rule Option '%s'\n",
             optionName);
     }
